components/storage/blockdevice/COMPONENT_SD/SDBlockDevice.cpp

```diff
@@ -1129,29 +1129,22 @@ void SDBlockDevice::_spi_wait(uint8_t count)
 
 void SDBlockDevice::_spi_init()
 {
-    _spi.lock();
     // Set to SCK for initialization, and clock card with cs = 1
     _spi.frequency(_init_sck);
     _spi.format(8, 0);
     _spi.set_default_write_value(SPI_FILL_CHAR);
     // Initial 74 cycles required for few cards, before selecting SPI mode
-    _cs = 1;
     _spi_wait(10);
-    _spi.unlock();
 }
 
 void SDBlockDevice::_select()
```

Comment:
I don't actually understand what is the purpose of this function. I would have assumed lock would be acquired here and released in __deselect_

Reply:
spi_wait() is:
// SPI function to wait for count
void SDBlockDevice::_spi_wait(uint8_t count)
{
    for (uint8_t i = 0; i < count; ++i) {
        _spi.write(SPI_FILL_CHAR);
    }
}
and _spi.write:
int SPI::write(int value)
{
select();
int ret = spi_master_write(&_peripheral->spi, value);
deselect();
return ret;
}

and deselect:
void SPI::deselect()
{
    if (--_select_count == 0) {
        _set_ssel(1);
    }
    unlock();
}
So unlock is in spi_wait.